### PR TITLE
[WIP] monkey vendor

### PR DESCRIPTION
**Pull request by timothyteakettle**
## About The Pull Request
monkey vendor
it shakes and makes monkey noises instead of advertising normally
dispenses live monkeys (they're not actually stored in the machine, it dispenses an abstract item that deletes itself and spawns a monkey)
it is **Very Funny**
haven't placed it anywhere on the map yet, but it'll probably go somewhere in medbay out of the way
i.e. genetics as a replacement to the monkey cube box they get

genetics are midway between science and medbay so the vendor is free for both science and medbay departments (dont wanna screw scientists over by removing one of their quick sources of early monkeys)

monkey vendor sprite done by Trip#8773 (known by name: Friend)

dependency: #15656

vending machine:
image
restock item:
image

## Why It's Good For The Game
ook

## Changelog
:cl:
add: adds a monkey vendor
/:cl:

### CODE REVIEW

File: code/modules/vending/monkey.dm [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	name = "vended monkey"
+	desc = "If you're seeing this, the coders made a mistake!"
+	item_flags = ABSTRACT // this stops it being placed into the user's hand by failing a can_put_in_hand check
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "valuechimp"
+
+/obj/item/vended_monkey/Initialize()
+	. = ..()
+	new /mob/living/carbon/monkey(get_turf(src))
+	qdel(src)
```
Comment (SandPoot) [inline code/modules/vending/monkey.dm]:
```suggestion
	return INITIALIZE_HINT_QDEL
```

**Comment by SandPoot** [comment]
Initialize hint was reverted

**Comment by timothyteakettle** [comment]
> Initialize hint was reverted

fixed thanks

**Comment by TripleShades** [comment]
Since the preview is a png, I'd like to point out that the vending machine does have an animation.
https://cdn.discordapp.com/attachments/580159267588538385/982027450424918076/valuechimp.gif

**Comment by shellspeed1** [comment]
Make it a randomly spawned vendor as a joke.

File: code/modules/vending/monkey.dm [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+// temporary monkey item that is dispensed onto the floor and then deletes itself, spawning a monkey
+/obj/item/vended_monkey
+	name = "vended monkey"
+	desc = "If you're seeing this, the coders made a mistake!"
+	item_flags = ABSTRACT // this stops it being placed into the user's hand by failing a can_put_in_hand check
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "valuechimp"
+
+/obj/item/vended_monkey/Initialize()
+	. = ..()
```
Comment (SandPoot) [inline code/modules/vending/monkey.dm:20]: Is it completely necessary to do a full init and then give it a hint to drop everything?

**Comment by DeltaFire15** [comment]
Dislike the existence of a whole vendor the entire purpose of which is a single semi-joke item which is just less convenient monkey cubes.

**Pull request closed without merging**